extensions/ql-vscode/src/query-history.ts

```diff
@@ -904,7 +913,37 @@ export class QueryHistoryManager extends DisposableObject {
         // If raw log does exist, then the summary log is still being generated.
         this.warnInProgressEvalLogSummary();
       } else {
-        this.warnNoEvalLogSummary();
+        this.warnNoEvalLogs();
+      }
+    }
+  }
+
+  async handleShowEvalLogVisualizer(
+    singleItem: QueryHistoryInfo,
+    multiSelect: QueryHistoryInfo[],
+  ) {
+    const { finalSingleItem, finalMultiSelect } = this.determineSelection(singleItem, multiSelect);
+    // Only applicable to an individual local query
+    if (!this.assertSingleQuery(finalMultiSelect) || !finalSingleItem || finalSingleItem.t !== 'local') {
+      return;
+    }
+
+    // If visualizer data in memory does exist, then build tree and display
+    if (finalSingleItem.evalLogVisualizerData) {
+      const evalLogTreeBuilder = new EvalLogTreeBuilder(finalSingleItem.evalLogVisualizerData);
```

Comment:
My previous comment about caching is not relevant yet, because I see we are constructing a fresh builder here each time we run the command. Let's not complicate it further, but there might be room to make this faster in future.